src/Azure/Orleans.Clustering.AzureStorage/OrleansSiloInstanceManager.cs

```diff
@@ -78,24 +78,21 @@ public void RegisterSiloInstance(SiloInstanceTableEntry entry)
         {
             entry.Status = INSTANCE_STATUS_CREATED;
             logger.Info(ErrorCode.Runtime_Error_100270, "Registering silo instance: {0}", entry.ToString());
-            storage.UpsertTableEntryAsync(entry)
-                .WaitWithThrow(AzureTableDefaultPolicies.TableOperationTimeout);
+            Task.WaitAll(new Task[] { storage.UpsertTableEntryAsync(entry) });
         }
 
         public void UnregisterSiloInstance(SiloInstanceTableEntry entry)
         {
             entry.Status = INSTANCE_STATUS_DEAD;
             logger.Info(ErrorCode.Runtime_Error_100271, "Unregistering silo instance: {0}", entry.ToString());
-            storage.UpsertTableEntryAsync(entry)
-                .WaitWithThrow(AzureTableDefaultPolicies.TableOperationTimeout);
+            Task.WaitAll(new Task[] { storage.UpsertTableEntryAsync(entry) });
```

Comment:
Why use wait all then there is one task?
Aren't we loosing timeout here?

Reply:
Addressed them in the new iteration